### DISCUSSION TITLE
sql attr max rows returns 0 value even we set some value

### DISCUSSION THREAD

**Discussion by kothapraveen** [FreeTDS/freetds#535] [Q&A] [answered]
Hi @brianb @freddy77 

Currently am working with below environment

unixODBC- 2.3.11
gcc540
RHEL7 64
FreeTDS 1.3

we are trying to validate the SQL_ATTR_MAX_ROWS using set & get methods
set method used SQLSetStmtAttrW() returns the SQL_SUCCESS_WITH_INFO
get method used SQLGetStmtAttrW returns SUCCESS

we are getting 0 value even we have set somevalue.
please observe below logs

[ODBC][116077][1710347356.702105][SQLSetStmtAttrW.c][166]
Entry:
Statement = 0xce73d0
Attribute = SQL_ATTR_MAX_ROWS
Value = 0x1f40
StrLen = -5
[ODBC][116077][1710347356.702193][SQLSetStmtAttrW.c][720]
Exit:[SQL_SUCCESS_WITH_INFO]
[ODBC][116077][1710347419.689357][SQLGetDiagRecW.c][535]
Entry:
Statement = 0xce73d0
Rec Number = 1
SQLState = 0x7ffff7dd58e0
Native = 0x7ffff7dd58ec
Message Text = 0x7ffff7dd54e0
Buffer Length = 511
Text Len Ptr = 0x7fffffff964e
DIAG [01S02] [FreeTDS][SQL Server]Option value changed

[ODBC][116077][1710347419.689501][SQLGetDiagRecW.c][596]
Exit:[SQL_SUCCESS]
SQLState = [01S02]
Native = 0x7ffff7dd58ec -> 0 (32 bits)
Message Text = [[FreeTDS][SQL Server]Option value changed]
[ODBC][116077][1710347419.689813][SQLGetDiagRecW.c][535]
Entry:
Statement = 0xce73d0
Rec Number = 2
SQLState = 0x7ffff7dd58e0
Native = 0x7ffff7dd58ec
Message Text = 0x7ffff7dd54e0
Buffer Length = 511
Text Len Ptr = 0x7fffffff964e
[ODBC][116077][1710347419.689861][SQLGetDiagRecW.c][596]
Exit:[SQL_NO_DATA]
[ODBC][116077][1710347445.296876][SQLGetStmtAttrW.c][153]
Entry:
Statement = 0xce73d0
Attribute = SQL_ATTR_MAX_ROWS
Value = 0x7fffffffb428
Buffer Length = -5
StrLen = 0x7fffffffb42c
[ODBC][116077][1710347445.296955][SQLGetStmtAttrW.c][348]
Exit:[SQL_SUCCESS]
Could you please some one can help on this issue

**Comment by freddy77** [comment, marked as answer]
The `Option value changed` message is the way of ODBC to tell you "I understand your request but I cannot use the value you passed so I changed it". 
Specifically for `SQL_ATTR_MAX_ROWS` the protocol has no way to tell the server the limit (note: the protocol, not the language) so it replaced it with 0.

**Reply by kothapraveen** [reply]
Thank you
